src/uct/ib/dc/base/dc_iface.h

```diff
@@ -48,8 +48,10 @@ typedef struct uct_dc_dci {
 
 typedef struct uct_dc_fc_sender_data {
     uint64_t                      ep;
-    int                           is_global;
-    union ibv_gid                 gid;
+    struct {
+        int                       is_global;
+        union ibv_gid             gid;
```

Comment:
maybe you can declare "union ibv_gid gid" as packed?

Reply:
uct_dc_fc_sender_data is sent in two arguments in mlx5 post_send: ep is passed as header, and 'global' as payload. to simplify arithmetic (pointer and sizeof) used nested structuring (and packed attribute)